src/frontend/planner_test/tests/testdata/cast.yaml

```diff
@@ -0,0 +1,90 @@
+# This file is automatically generated. See `src/frontend/planner_test/README.md` for more information.
+- name: implicit cast boolean (HAVING NULL)
+  sql: |
+    select 1 having NULL;
+  batch_plan: |
+    BatchProject { exprs: [1:Int32] }
+    └─BatchFilter { predicate: null:Boolean }
+      └─BatchSimpleAgg { aggs: [] }
+        └─BatchValues { rows: [[]] }
+- name: implicit cast boolean (WHERE NULL)
+  sql: |
+    select 1 where NULL;
+  batch_plan: |
+    BatchProject { exprs: [1:Int32] }
+    └─BatchFilter { predicate: null:Boolean }
+      └─BatchValues { rows: [[]] }
+- name: implicit cast boolean (CASE NULL)
+  sql: |
+    select case when NULL then 1 end;
+  batch_plan: |
+    BatchProject { exprs: [Case(null:Boolean, 1:Int32)] }
+    └─BatchValues { rows: [[]] }
+- name: implicit cast boolean (JOIN ON NULL)
+  sql: "create table t1(v1 int); \ncreate table t2(v1 int); \nselect t1.v1 from t1 join t2 on NULL;\n"
```

Comment:
Is there some way to avoid this being rewritten to this form?

Reply:
Just edited directly. No reformat after run `do-apply-planner-test` again. I guess it doesn't edit if output is as expected.

Reply:
IIRC it is due to the trailing space before `\n`.